### PR TITLE
Run3-alca243 Add the ioread statement to read back old HOCalibVariables in the new cases

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Add the ioread statement to read back old HOCalibVariables in the new cases

#### PR validation:

No specific test is done

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

File: DataFormats/HcalCalibObjects/src/classes_def.xml [hunk 1]
```diff
@@ -3,6 +3,9 @@
    <version ClassVersion="16" checksum="2161587371"/>
    <version ClassVersion="17" checksum="1351689277"/>
   </class>
+   <ioread sourceClass="HOCalibVariables" targetClass="HOCalibVariables" version="[-16]" target="pileup" source="double insLumi" embed="false">
+    <![CDATA[ pileup = onfile.insLumi; ]]>
+  </ioread>
```
Comment (missirol) [inline DataFormats/HcalCalibObjects/src/classes_def.xml:8]:
```suggestion
   <ioread sourceClass="HOCalibVariables" targetClass="HOCalibVariables" version="[-16]" target="pileup" source="float inslumi" embed="false">
    <![CDATA[ pileup = onfile.inslumi; ]]>
  </ioread>
```
I'd assume ioread rules are case-sensitive (also, the type was `float`).
https://github.com/bsunanda/cmssw/commit/ef5ad54a923bb574888b3373357cd931ab441fd6#diff-05559b5e9d212b1045ba8feaed01c0f7b24c190f79b82807e02cadb0e052c2edL10

**Comment by perrotta** [comment]
type bug-fix
(I imagine)

**Comment by tvami** [comment]
+alca

**Comment by perrotta** [comment]
hold
- To be answered/fixed https://github.com/cms-sw/cmssw/pull/41811#discussion_r1210559082

**Comment by tvami** [comment]
> > +alca
> 
> how about [#41811 (comment)](https://github.com/cms-sw/cmssw/pull/41811#discussion_r1210559082) ?

indeed, sorry for signing early, I removed the signature now

**Comment by bsunanda** [comment]
@cmsbuild please test

**Comment by perrotta** [comment]
unhold

**Comment by tvami** [comment]
@bsunanda please squash

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by tvami** [comment]
+alca
 * last push is just a squash

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**